feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go

```diff
@@ -449,58 +454,103 @@ func TestTraffic(t *testing.T) {
+
+	t.Logf("Configuring traffic from ATE through DUT...")
+
+	v4Flow := ts.ATETop.Flows().Add()
+	v4Flow.SetName("v4Flow")
+	v4Flow.TxRx().Device().SetTxNames([]string{srcIpv4.Name()}).SetRxNames([]string{netv4.Name()})
+
+	v4FlowEth := v4Flow.Packet().Add().Ethernet()
+	v4FlowEth.Src().SetValue(session.ATETrafficAttrs.MAC)
+
+	v4FlowIp := v4Flow.Packet().Add().Ipv4()
+	v4FlowIp.Src().SetValue(session.ATETrafficAttrs.IPv4)
+	v4FlowIp.Dst().SetValue(targetNetwork.IPv4)
+
+	// v4Flow.Duration().FixedPackets().SetPackets(100)
+	v4Flow.Rate().SetPps(50)
+	v4Flow.Size().SetFixed(128)
+	v4Flow.Metrics().SetEnable(true)
+
+	srcIpv6 := srcIntf.Ethernets().Items()[0].Ipv6Addresses().Items()[0]
+	v6Flow := ts.ATETop.Flows().Add()
+	v6Flow.SetName("v6Flow")
+	v6Flow.TxRx().Device().SetTxNames([]string{srcIpv6.Name()}).SetRxNames([]string{netv6.Name()})
+
+	v6FlowEth := v6Flow.Packet().Add().Ethernet()
+	v6FlowEth.Src().SetValue(session.ATETrafficAttrs.MAC)
+
+	v6FlowIp := v6Flow.Packet().Add().Ipv6()
+	v6FlowIp.Src().SetValue(session.ATETrafficAttrs.IPv6)
+	v6FlowIp.Dst().SetValue(targetNetwork.IPv6)
+
+	// v6Flow.Duration().FixedPackets().SetPackets(100)
+	v6Flow.Rate().SetPps(50)
+	v6Flow.Size().SetFixed(128)
+	v6Flow.Metrics().SetEnable(true)
+
+	deadFlow := ts.ATETop.Flows().Add()
+	deadFlow.SetName("deadFlow")
+	deadFlow.TxRx().Device().SetTxNames([]string{srcIpv4.Name()}).SetRxNames([]string{netv4.Name()})
+
+	deadFlowEth := deadFlow.Packet().Add().Ethernet()
+	deadFlowEth.Src().SetValue(session.ATETrafficAttrs.MAC)
+
+	deadFlowIp := deadFlow.Packet().Add().Ipv4()
+	deadFlowIp.Src().SetValue(session.ATETrafficAttrs.IPv4)
+	deadFlowIp.Dst().SetValue(deadNetwork.IPv4)
+
+	// deadFlow.Duration().FixedPackets().SetPackets(100)
+	deadFlow.Rate().SetPps(50)
+	deadFlow.Size().SetFixed(128)
+	deadFlow.Metrics().SetEnable(true)
+
 	t.Logf("Starting protocols on ATE...")
 	ts.PushAndStart(t)
-	defer ts.ATETop.StopProtocols(t)
 	ts.MustAdjacency(t)
-	t.Logf("Configuring traffic from ATE through DUT...")
-	v4Header := ondatra.NewIPv4Header()
-	v4Header.DstAddressRange().WithMin(targetNetwork.IPv4).WithCount(1)
-	v4Flow := ate.Traffic().NewFlow("v4Flow").
-		WithSrcEndpoints(srcIntf).WithDstEndpoints(dstIntf).
-		WithHeaders(ondatra.NewEthernetHeader(), v4Header)
-	v6Header := ondatra.NewIPv6Header()
-	v6Header.DstAddressRange().WithMin(targetNetwork.IPv6).WithCount(1)
-	v6Flow := ate.Traffic().NewFlow("v6Flow").
-		WithSrcEndpoints(srcIntf).WithDstEndpoints(dstIntf).
-		WithHeaders(ondatra.NewEthernetHeader(), v6Header)
-	// deadFlow is addressed to a nonexistent network as a consistency check -
-	// all traffic should be blackholed.
-	deadHeader := ondatra.NewIPv4Header()
-	deadHeader.DstAddressRange().WithMin(deadNetwork.IPv4).WithCount(1)
-	deadFlow := ate.Traffic().NewFlow("flow2").
-		WithSrcEndpoints(srcIntf).WithDstEndpoints(dstIntf).
-		WithHeaders(ondatra.NewEthernetHeader(), deadHeader)
+
+	time.Sleep(time.Second * 30)
```

Comment:
Sleep statement got duplicated?

Reply:
Replaced with an otg telemetry call to wait until the first prefix is received. Without this, the traffic won't start. In the future "PrefixAny()." will be replaced with "Prefix(targetNetwork.IPv4)" once this becomes available in otg

Reply:
Sleep was replaced with a otgtelemetry command to check for the incoming IS-IS L2 LSP. At some point the TODO section will be implemented as is more accurate.